### ISSUE TITLE
Cannot pick KEY from nested Avro field

### ISSUE THREAD

**Issue by SamirHafez** [confluentinc/ksql#1975]
KSQL introduced the ability to handle nested structures for Avro records. The problem in when creating a TABLE, the `KEY` field provided in the `WITH` clause doesn't support nested fields.

Having the structure (pseudo-avro):
```
{"name": "hello!",
 "inner": {"id", "12345"}}
```
It's not possible to do:
`CREATE TABLE <X> WITH (..., KEY='inner.id')`

To get around this atm, it would be required to create a STREAM from the topic, then another STREAM which surfaces the required field to the root of the structure. Then it would be possible to simply use it for creating a TABLE;

**Comment by miguno** [comment]
And to add to this:  It's not just Avro, it's also a problem for JSON afaict.

**Comment by rmoff** [comment]
This should be addressed in the wider issue here https://github.com/confluentinc/ksql/issues/824
